doc/GUIDE.md

```diff
@@ -85,6 +85,27 @@ function pointer types, and typedefs of struct or union types result in addition
 being generated. (All of these are discussed in more detail in the
 [Using The Code Generated By Jextract section](#using-the-code-generated-by-jextract)).
 
+Generally speaking, the code that jextract generates is dependent on the platform on which
+jextract is running. For example, when a C header file is processed by the C pre-processor,
+it is possible for code in the header file to detect the current platform using
+pre-processor directives, and expand to a different result based on that platform (see
+also the section on [pre-processor definitions](#preprocessor-definitions)). Additionally,
+different built in C types can have different formats depending on the platform (for
+example, the `long` type has different formats on Linux and Windows). Both of these are
```

Comment:
Maybe we also want to mention padding in structs?

Reply:
That only really comes into play on ABIs that use a different packing (like the AIX ABI), so it doesn't really seem that important to mention. Any way, the things I mentioned are just a couple of examples. With pre-processor directives a header could really create any kind of platform dependency it wants.